tests/jax/integration/parameter_server_test.py

```diff
@@ -1,89 +1,87 @@
-        env_name="simple_spread",
-        action_space="discrete",
-    )
-
-    # Networks.
-    network_factory = ippo.make_default_networks
-
-    test_system.build(
-        environment_factory=environment_factory,
-        network_factory=network_factory,
-        non_blocking_sleep_seconds=0,
-    )
+def test_system_sp() -> System:
+    """A single process built system"""
+    return mock_system_single_process()
+
+
+def test_parameter_server_single_process(test_system_sp: System) -> None:
+    """Test if the paraameter server instantiates processes as expected."""
     (
         data_server,
         parameter_server,
         executor,
         evaluator,
         trainer,
-    ) = test_system._builder.store.system_build
-    assert type(parameter_server) == ParameterServer
+    ) = test_system_sp._builder.store.system_build
+
+    # Initial state of the parameter_server
+    assert type(parameter_server.store.system_checkpointer) == savers.Checkpointer
 
-    step_var = parameter_server.get_parameters("trainer_steps")
-    assert type(step_var) == np.ndarray
-    assert step_var[0] == 0
+    for network in parameter_server.store.parameters["networks-network_agent"].values():
+        assert sorted(list(network.keys())) == ["b", "w"]
+        assert jnp.size(network["w"]) != 0
+        assert jnp.size(network["b"]) != 0
 
-    parameter_server.set_parameters({"trainer_steps": np.ones(1, dtype=np.int32)})
-    assert parameter_server.get_parameters("trainer_steps")[0] == 1
+    param_without_net = parameter_server.store.parameters.copy()
+    del param_without_net["networks-network_agent"]
+    assert param_without_net == {
+        "trainer_steps": jnp.zeros(1, dtype=jnp.int32),
+        "trainer_walltime": jnp.zeros(1, dtype=jnp.float32),
+        "evaluator_steps": jnp.zeros(1, dtype=jnp.int32),
+        "evaluator_episodes": jnp.zeros(1, dtype=jnp.int32),
+        "executor_episodes": jnp.zeros(1, dtype=jnp.int32),
+        "executor_steps": jnp.zeros(1, dtype=jnp.int32),
+    }
 
-    parameter_server.add_to_parameters({"trainer_steps": np.ones(1, dtype=np.int32)})
-    assert parameter_server.get_parameters("trainer_steps")[0] == 2
+    assert not parameter_server.store.last_checkpoint_time
+    assert parameter_server.store.system_checkpointer._last_saved == 0
 
-    # Step the parameter sever
+    # test get and set parameters
+    for _ in range(3):
+        executor.run_episode()
+    trainer.step()
+
+    trainer_steps = parameter_server.get_parameters("trainer_steps")
+    executor_episodes = parameter_server.get_parameters("executor_episodes")
+    assert list(trainer_steps) == [1]  # trainer.step one time
+    assert list(executor_episodes) == [3]  # run episodes three times
+
+    network = parameter_server.get_parameters("networks-network_agent")
+    for network in network.values():
+        assert sorted(list(network.keys())) == ["b", "w"]
+        assert jnp.size(network["w"]) != 0  # network is updated
```

Comment:
Does this not just test that the networks are initialised and not really test for updating?

Reply:
but aren't we run the episode and the trainer so it w ill be updated?

Reply:
But we are not testing for updating? We are testing if it is not zero? It is not zero after initialisation as well. Or am I misunderstanding the test?

Reply:
oh I see that make sense my bad I will make previous network and see if it is updated

Reply:
Done I updated the test

Reply:
Great, thanks! :slightly_smiling_face: